File: ui/src/app/model/UserDashboardVO.ts
import { IncidentModuleVO } from "./IncidentModuleVO";
import { IncidentStatusVO } from "./IncidentStatusVO";

export class UserDashboardVO {
    moduleWiseIncidents: IncidentModuleVO[];
    statusWiseIncidents: IncidentStatusVO[];
    moduleWiseAssetIncidents: IncidentModuleVO[];
    statusWiseAssetIncidents: IncidentStatusVO[];
    constructor(data: any) {
        if (data == null) {
            this.moduleWiseIncidents = [];
            this.statusWiseIncidents = [];
            this.moduleWiseAssetIncidents = [];
            this.statusWiseAssetIncidents = [];
        } else {
            this.moduleWiseIncidents = data.moduleWiseIncidents;
            this.statusWiseIncidents = data.statusWiseIncidents;
            this.moduleWiseAssetIncidents = data.moduleWiseAssetIncidents;
            this.statusWiseAssetIncidents = data.statusWiseAssetIncidents;
        }
    }
}

File: domain/src/main/java/com/centram/common/vo/UserDashboardVO.java
package com.centram.common.vo;


import lombok.*;

import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
public class UserDashboardVO {
    private Set<IncidentModuleVO> moduleWiseIncidents;
    private Set<IncidentStatusVO> statusWiseIncidents;
    private Set<IncidentModuleVO> moduleWiseAssetIncidents;
    private Set<IncidentStatusVO> statusWiseAssetIncidents;
}